sdk/include/xdk/ketypes.h

```diff
@@ -1281,30 +1281,67 @@ typedef struct _KUSER_SHARED_DATA
     ULONG CryptoExponent;                                   // 0x23c
     ULONG TimeZoneId;                                       // 0x240
     ULONG LargePageMinimum;                                 // 0x244
+
+#if (NTDDI_VERSION >= NTDDI_WIN8)
+    ULONG AitSamplingValue;                                 // 0x248
+    ULONG AppCompatFlag;                                    // 0x24c
+    ULONGLONG RNGSeedVersion;                               // 0x250
+    ULONG GlobalValidationRunlevel;                         // 0x258
+    volatile LONG TimeZoneBiasStamp;                        // 0x25c
+#if (NTDDI_VERSION >= NTDDI_WIN10)
+    ULONG NtBuildNumber;                                    // 0x260
+#else
+    ULONG Reserved2;                                        // 0x260
+#endif
+#else
     ULONG Reserved2[7];                                     // 0x248
+#endif // NTDDI_VERSION >= NTDDI_WIN8
+
     NT_PRODUCT_TYPE NtProductType;                          // 0x264
     BOOLEAN ProductTypeIsValid;                             // 0x268
+    BOOLEAN Reserved0[1];                                   // 0x269
+#if (NTDDI_VERSION >= NTDDI_WIN8)
+    USHORT NativeProcessorArchitecture;                     // 0x26a
+#endif
     ULONG NtMajorVersion;                                   // 0x26c
     ULONG NtMinorVersion;                                   // 0x270
     BOOLEAN ProcessorFeatures[PROCESSOR_FEATURE_MAX];       // 0x274
     ULONG Reserved1;                                        // 0x2b4
     ULONG Reserved3;                                        // 0x2b8
     volatile ULONG TimeSlip;                                // 0x2bc
     ALTERNATIVE_ARCHITECTURE_TYPE AlternativeArchitecture;  // 0x2c0
+#if (NTDDI_VERSION >= NTDDI_WIN10)
+    ULONG BootId;                                           // 0x2c4
+#else
     ULONG AltArchitecturePad[1];                            // 0x2c4
+#endif
     LARGE_INTEGER SystemExpirationDate;                     // 0x2c8
     ULONG SuiteMask;                                        // 0x2d0
     BOOLEAN KdDebuggerEnabled;                              // 0x2d4
-#if (NTDDI_VERSION >= NTDDI_WINXPSP2)
-    UCHAR NXSupportPolicy;                                  // 0x2d5
+    union
+    {
+        UCHAR MitigationPolicies;                           // 0x2d5
+        struct
+        {
+            UCHAR NXSupportPolicy : 2;
+            UCHAR SEHValidationPolicy : 2;
+            UCHAR CurDirDevicesSkippedForDlls : 2;
+            UCHAR Reserved : 2;
+        };
+    };
+#if (NTDDI_VERSION >= NTDDI_WIN10)
+    USHORT CyclesPerYield;                                  // 0x2d6 // Win 10 19H1+
```

Comment:
Incidentally we have the `NTDDI_WIN10_19H1` define, which you may find useful here.
(see https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=define+NTDDI_WIN10 )

Reply:
Fixed